tests/apollo/util/bft.py

```diff
@@ -1314,24 +1407,30 @@ async def wait_for_state_transfer_to_stop(self, up_to_date_node, stale_node, sto
                         with trio.move_on_after(.5): # seconds
                             metrics = await self.metrics.get_all(stale_node)
                             try:
-                                n = self.metrics.get_local(metrics, *key)
+                                stale_node_metric = self.metrics.get_local(metrics, *key)
                             except KeyError:
                                 # ignore - the metric will eventually become available
                                 await trio.sleep(0.1)
                             else:
                                 # Debugging
-                                if n != last_n:
-                                    last_n = n
+                                if stale_node_metric != last_n:
+                                    last_n = stale_node_metric
                                     last_stored_checkpoint = self.metrics.get_local(metrics,
                                         'bc_state_transfer', 'Gauges', 'last_stored_checkpoint')
                                     on_transferring_complete = self.metrics.get_local(metrics,
                                         'bc_state_transfer', 'Counters', 'on_transferring_complete')
                                     action.log(message_type="Not complete yet",
-                                        seq_num=n, last_stored_checkpoint=last_stored_checkpoint, on_transferring_complete=on_transferring_complete)
-
-                                # Exit condition
-                                if n >= expected_seq_num:
-                                    action.add_success_fields(n=n, expected_seq_num=expected_seq_num)
+                                               seq_num=stale_node_metric, expected_seq_num=expected_seq_num,
+                                               last_stored_checkpoint=last_stored_checkpoint,
+                                               on_transferring_complete=on_transferring_complete)
+                                    log.log_message(message_type="Not complete yet",
```

Comment:
what is the difference between `log.log_message` and `action.log`?
Do we need them both?

Reply:
We don't need both.